Rerender

ocefpaf opened the pull request:
This is pulling `hdf5 1.10.5` :-/

ocefpaf committed:
MNT: Re-rendered with conda-build 3.18.9, conda-smithy 3.4.1, and conda-forge-pinning 2019.07.23

ocefpaf committed:
bump

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

minrk committed:
list hdf5 twice in host deps

need to list hdf5 twice to get version pinning from conda_build_config (`hdf5` no version)
and build pinning from `{{ mpi_prefix }}` (`hdf5 * {{ mpi_prefix }}`).

Alternately, could specify version in host dep directly instead of conda_build_config.py

minrk commented:
Apparently this dep:

```yaml
- hdf5 * {{ mpi_prefix }}
```

is overriding the hdf5 version pinning in conda_build_config.yaml as explicitly allowing any (and thereby implicitly pulling in latest). Didn't come up before, because we were pinning latest. Adding an extra `- hdf` dep works for me locally with `conda render`, so hoping that works.

Not sure if this is a conda-build bug or not.

bgruening commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

bgruening commented:
Thanks this seems to work.

ax3l commented on recipe/meta.yaml:
```diff
@@ -41,6 +41,9 @@ requirements:
     - mpi4py  # [mpi != 'nompi']
+    # need to list hdf5 twice to get version pinning from conda_build_config
+    # and build pinning from {{ mpi_prefix }}
+    - hdf5
```
@minrk do you think this work-around should be applied to all mpi-varianted packages?
How can one verify one is affected? I usually specify min-version here, so that is fine?
(e.g. `adios`, `openpmd-api` feedstocks)

minrk replied:
Yeah, I think so
